src/java.base/share/classes/sun/security/x509/NameConstraintsExtension.java

```diff
@@ -127,16 +127,20 @@ private void encodeThis() throws IOException {
     }
 
     /**
-     * The default constructor for this class. Both parameters
-     * are optional and can be set to null.  The extension criticality
+     * The default constructor for this class. Both parameters are optional
+     * but at least one should be non null.  The extension criticality
      * is set to true.
      *
      * @param permitted the permitted GeneralSubtrees (null for optional).
      * @param excluded the excluded GeneralSubtrees (null for optional).
      */
     public NameConstraintsExtension(GeneralSubtrees permitted,
                                     GeneralSubtrees excluded)
-    throws IOException {
+            throws IOException {
+        if (permitted == null && excluded == null) {
+            throw new IllegalArgumentException(
+                    "permitted and exclude cannot both be null");
```

Comment:
s/exclude/excluded/

Reply:
Do you want to add similar checks to the `GeneralSubtree` and `GeneralSubtrees` ctors?

Reply:
Ok.